packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -208,17 +209,19 @@ class Viewport implements IViewport {
 
     // If the pan has been applied, we need to be able
     // apply the pan back
-    const resetFocalPoint = vec3.create();
     const dimensions = imageData.getDimensions();
     const middleIJK = dimensions.map((d) => Math.floor(d / 2));
 
+    const centeredFocalPoint = vec3.create();
     const idx = [middleIJK[0], middleIJK[1], middleIJK[2]];
-    imageData.indexToWorld(idx, resetFocalPoint);
+    imageData.indexToWorld(idx, centeredFocalPoint);
+
+    const resetFocalPoint = this._getFocalPointForResetCamera(
+      centeredFocalPoint as Point3,
```

Comment:
You don't need the extra variable - just use vec3.create() inline here.
Also, consider making the definition for Point3 as being identical to the vec3 one so that no casting is required.  I know that means you can inadvertently pass a wrong length FloatArray32, but anything using regular arrays will be type checked, and the others are mostly created by the vec3 functions and as cast, so it doesn't decrease type testing.

Reply:
I always forget that :D